packages/notebook-extension/src/index.ts

```diff
@@ -1572,6 +1581,70 @@ function activateNotebookHandler(
   return tracker;
 }
 
+/**
+ * Activate the completer service for notebook.
+ */
+function activateNotebookCompleterService(
+  app: JupyterFrontEnd,
+  notebooks: INotebookTracker,
+  manager?: ICompletionProviderManager
+): void {
+  if (!manager) {
+    return;
+  }
+  const invokeNotebook = 'completer:invoke-notebook';
+  const selectNotebook = 'completer:select-notebook';
+  app.commands.addCommand(invokeNotebook, {
+    execute: args => {
+      const panel = notebooks.currentWidget;
+      if (panel && panel.content.activeCell?.model.type === 'code') {
+        manager.invoke(panel.id);
+      }
+    }
+  });
+
+  app.commands.addCommand(selectNotebook, {
+    execute: () => {
+      const id = notebooks.currentWidget && notebooks.currentWidget.id;
+
+      if (id) {
+        return manager.select(id);
+      }
+    }
+  });
+
+  app.commands.addKeyBinding({
+    command: selectNotebook,
+    keys: ['Enter'],
+    selector: '.jp-Notebook .jp-mod-completer-active'
+  });
+
+  notebooks.widgetAdded.connect(async (_, notebook) => {
+    const completerContext = {
+      editor: notebook.content.activeCell?.editor ?? null,
+      session: notebook.sessionContext.session,
+      widget: notebook
+    };
+    await manager.attachWidget(notebook, completerContext);
```

Comment:
There is an inconsistency in the API the widget is passed twice: once as first argument and in the context object.

You should pass it only once. And to align with most JLab API I would keep only the object as single argument for `attachWidget`.

> This likely means that `widget` is not optional but mandatory.